hist/hist/src/TH1.cxx

```diff
@@ -9430,6 +9431,96 @@ TH1* TH1::TransformHisto(TVirtualFFT *fft, TH1* h_output,  Option_t *option)
+
+   // Configure all axes
+   for (size_t i = 0; i < ndim; ++i) {
+      const auto &axis = (i == 0) ? fXaxis : (i == 1) ? fYaxis : fZaxis;
+      Int_t binLow = std::max(1, args[i * 2]);
+      Int_t binUp = std::min(axis.GetNbins() + 1, args[i * 2 + 1]);
+      args[i * 2] = binLow;
+      args[i * 2 + 1] = binUp;
+      nBins[i] = binUp - binLow;
+      binLowEdge[i] = axis.GetBinLowEdge(binLow);
+      binUpEdge[i] = axis.GetBinLowEdge(binUp);
+      shiftedBins[i * 2] = 1;
+      shiftedBins[i * 2 + 1] = nBins[i] + 1;
+   }
+
+   // Create a new histogram of the same type as this
+   std::unique_ptr<TH1> hSlice(static_cast<TH1*>(IsA()->GetNew()(nullptr)));
+
+   if (!hSlice) {
+      throw std::runtime_error("Failed to create a new histogram instance.");
+   }
+
+   // Configure name, title and bins
+   hSlice->SetNameTitle(Form("%s_slice", GetName()), GetTitle());
+
+   if (ndim == 1) {
+      hSlice->SetBins(nBins[0], binLowEdge[0], binUpEdge[0]);
+   } else if (ndim == 2) {
+      hSlice->SetBins(nBins[0], binLowEdge[0], binUpEdge[0],
+                  nBins[1], binLowEdge[1], binUpEdge[1]);
+   } else if (ndim == 3) {
+      hSlice->SetBins(nBins[0], binLowEdge[0], binUpEdge[0],
+                  nBins[1], binLowEdge[1], binUpEdge[1],
+                  nBins[2], binLowEdge[2], binUpEdge[2]);
+   }
+
+   std::vector<Double_t> sliceContents;
+   std::unordered_set<Int_t> processedBins;
+   Double_t underflowContent = 0.0, overflowContent = 0.0;
+
+   auto processBinContent = [&](Int_t x, Int_t y, Int_t z) {
+      const Int_t globalBin = GetBin(x, y, z);
+      if (!processedBins.insert(globalBin).second) return;
+
+      const auto content = RetrieveBinContent(globalBin);
+      const bool isUnderflow = (x < args[0] || (ndim > 1 && y < args[2]) || (ndim > 2 && z < args[4]));
+      const bool isOverflow = (x >= args[1] || (ndim > 1 && y >= args[3]) || (ndim > 2 && z >= args[5]));
+
+      if (isUnderflow) {
+          underflowContent += content;
+      } else if (isOverflow) {
+          overflowContent += content;
+      } else {
+          sliceContents.push_back(content);
+      }
+   };
+
+   for (Int_t x = 0; x <= fXaxis.GetNbins() + 1; ++x) {
+      for (Int_t y = 0; y <= (ndim > 1 ? fYaxis.GetNbins() + 1 : 1); ++y) {
+          for (Int_t z = 0; z <= (ndim > 2 ? fZaxis.GetNbins() + 1 : 1); ++z) {
+              processBinContent(x, y, z);
+          }
+      }
+  }
+
+   // Set the contents of the slice histogram
+   ROOT::Internal::SetSliceContent(*hSlice, sliceContents, shiftedBins);
+   
+   // Set the flow bins
+   hSlice->SetBinContent(0, underflowContent);
+   auto overflowBin = (ndim == 1) ? hSlice->GetBin(fXaxis.GetNbins() + 1)
+                  : (ndim == 2) ? hSlice->GetBin(fXaxis.GetNbins() + 1, fYaxis.GetNbins() + 1)
+                  : hSlice->GetBin(fXaxis.GetNbins() + 1, fYaxis.GetNbins() + 1, fZaxis.GetNbins() + 1);
+   hSlice->SetBinContent(overflowBin, overflowContent);
+
+   return hSlice.release();
```

Comment:
I also don't understand why we need to release the pointer here, can't we just return a unique_ptr?